bfsbverify

```diff
@@ -63,7 +63,7 @@ debug=${DEBUG_ENABLE:-}
 
 scratch=${SCRATCH_PATH:-/tmp}
 # Indirection to protect against "rm -fr /" mistakes.
-tmpdir=$scratch/bfbtmp
+tmpdir=$scratch/bfbtmp-$USER
```

Comment:
Thanks for the comment Liming. It is good to know about 'mktemp -d' usage in other scripts.

I would like to have something very easy to manage and identify (especially when debugging/inspecting the command temp files). Also I would like to avoid creating temp directories in current location (other than /tmp/).

It looks like 'mktemp -d' command with other options such ad '-p' allows to create unique temporary directories with a specific template (directories can be created under /tmp). BUT still it is something the user (eventually) has to manage if running the command from a common server such as a farm machine or a lab machine. In case the command couldn't run to completion the temp directory will be kept there unless the user explicitly deletes it. The next command won't do it as it creates a new temp folder. For that reason I thought maybe a user-based temp file would be the best option here; especially while running on common resources (where servers aren't rebooted and the /tmp isn't cleaned-up that often).

What do you think?

Reply:
Would something like  tmpdir=$(mktemp -d $scratch/bfbtmp-XXXX) be what you need?
It'll be the same as tmpdir=$scratch/bfbtmp. The only difference is that it generate random name instead of using $USER

Reply:
The reason is that multiple people might run this command as root, so the problem is still there if using $USER

Reply:
Command executed as root can delete the temp directory, no issues with that.

The issue is with test commands or incomplete commands which leaves temp folders. Let's discuss offline. I can provide exact use cases.

Reply:
I can think of a combination of both too $(mktemp -d $scratch/bfbtmp-$USER.XXXX). But there is no point of having unique directories per user unless everyone is debugging the command simultaneously as root or any other common user.  

Again my concern is getting /tmp full on common servers (not talking about BlueField, I am talking about x86 servers). Because if the command is interrupted multiple times; it leaves the temp folder in /tmp. Unless the user manually deletes it. On the other hand if people are running the command as root then no issues the command will always cleanup the temp folder. Same for an individual user; every new command cleans up the temp folder.